pkgs/wldash/default.nix

```diff
@@ -0,0 +1,37 @@
+{ lib, rustPlatform, fetchFromGitHub, pkgconfig, dbus, libpulseaudio, libxkbcommon, wayland }:
+
+let
+  metadata = import ./metadata.nix;
+  libraryPath = lib.makeLibraryPath [ wayland libxkbcommon ];
+in
+rustPlatform.buildRustPackage rec {
+  name = "wldash-${version}";
+  version = metadata.rev;
+
+  src = fetchFromGitHub {
+    owner = "kennylevinsen";
+    repo = "wldash";
+    rev = metadata.rev;
+    sha256 = metadata.sha256;
+  };
+
+  cargoSha256 = "05sz4xpk9a7jhpis4ac9bwxy77w6q5xkp482wlzh87w52ixmakh6";
+
+  nativeBuildInputs = [ pkgconfig ];
+
+  buildInputs = [ dbus libpulseaudio ];
+
+  dontPatchELF = true;
+  
+  postInstall = ''
+    patchelf --set-rpath ${libraryPath}:$(patchelf --print-rpath $out/bin/wldash) $out/bin/wldash
```

Comment:
Why is the `patchelf` necessary? I'm not accustomed to seeing binaries built with `buildRustPackage` needing it? Does it re-execute itself or something?

Reply:
It seems to search for the wayland libraries dynamically. It's caused by the "dlopen" flag in the `cargo.toml` file. If I patched that out it could look for the wayland libraries at build time at which point I could just pass it in as a buildInput but I couldn't find a way to do this for libxkb.

 In the end I thought it was neater to patch the executable than start patching everything though I am willing to believe that I am totally wrong and there is a far better way to do this as it does seem strange